File: 07-interfaces/java/Mamifero.java
/*
 * Mamifero.java - SEGUNDA RAMA de la jerarquia
 *
 * Un Mamifero y un Ave no tienen NADA en comun salvo ser Animal.
 * Y sin embargo, un delfin y un pato saben hacer lo mismo: nadar.
 * Ese "y sin embargo" es la razon de existir de las interfaces.
 */
public abstract class Mamifero extends Animal {

    protected boolean tienePelaje;

    public Mamifero(String nombre, int edad, boolean tienePelaje) {
        super(nombre, edad);
        this.tienePelaje = tienePelaje;
    }

    @Override
    public String getGrupo() {
        return "Mamifero";
    }
}

File: 07-interfaces/python/mamifero.py
"""
mamifero.py - SEGUNDA RAMA de la jerarquia

Un Mamifero y un Ave no tienen NADA en comun salvo ser Animal.
Y sin embargo, un delfin y un pato saben hacer lo mismo: nadar.
Ese "y sin embargo" es la razon de existir de las interfaces.
"""

from animal import Animal


class Mamifero(Animal):

    def __init__(self, nombre, edad, tiene_pelaje):
        super().__init__(nombre, edad)
        self._tiene_pelaje = tiene_pelaje

    def get_grupo(self):
        return "Mamifero"
